src/model/core.py

```diff
@@ -2,25 +2,59 @@
 import copy
 import numpy as np
 
-def erode(im, size):
-    eroded = copy.copy(im)
-    if size > 1:
-        for i in range(len(im)):
-            eroded[i] = morphology.erosion(im[i], morphology.square(size))
+def erode(im, size, round_shape=True):
+    """
+    return an eroded image
+    im: 2d or 3d array
+    size:   int:    generate an element of size (size*2+1)
+    round_shape: str:    if True the elemnt is ball or disk
+    """
+    if size == 0:
+        return im
+    if len(im.shape) == 3:
+        if round_shape:
+            selem = morphology.ball(size)
+        else:
+            selem = morphology.cube(size*2+1)
+    elif len(im.shape) == 2:
+        if round_shape:
+            selem = morphology.disk(size)
+        else:
+            selem = morphology.square(size*2+1)
+
+    eroded = morphology.erosion(im, selem)
     return eroded
 
-def dilate(im, size):
-    dilated = copy.copy(im)
-    if size > 1:
-        for i in range(len(im)):
-            dilated[i] = morphology.dilation(im[i], morphology.square(size))
+def dilate(im, size, round_shape=True):
+    """
+    return a dilated image
+    im: 2d or 3d array
+    round_shape: str:    if True the elemnt is ball or disk
+    size:   int:    generate an element of size (size*2+1)
+    """
+    if size == 0:
+        return im
+    if len(im.shape) == 3:
+        if round_shape:
+            selem = morphology.ball(size)
+        else:
+            selem = morphology.cube(size*2+1)
+    elif len(im.shape) == 2:
+        if round_shape:
+            selem = morphology.disk(size)
+        else:
+            selem = morphology.square(size*2+1)
+    dilated = morphology.dilation(im, selem)
     return dilated
 
-
 def applyThreshold(im, threshold, reverse=False):
-    mask = np.zeros_like(im)
+    """
+    return a binary image
+    im: ndarray
+    threshold:  int or float:   pixel value
+    """
     if reverse:
-        mask[im < threshold] = 255
+        mask = im < threshold
     else:
-        mask[im > threshold] = 255
-    return mask
+        mask = im > threshold
+    return mask.astype(np.uint8)*255
```

Comment:
Threshold return a binary image, maybe you can return an image with 0 or 1, instead of 0 or 255?

Reply:
Yes we should. 
This modification has an effect on the view so I will had an argument 'cmap' to the node class to be able to associate each pixel value to a specified color. Otherwise a binary image using the defaut grayscale cmap cannot be visualize.  
Where do I commit this modification?

Reply:
You can commit this modification here